Add FreeBSD to README (#311)
liquidctl/liquidctl: README.md

[bash completions]: extra/completions/liquidctl.bash

## Installing on FreeBSD

liquidctl is maintained in the Ports Collection (thanks to ehaupt@FreeBSD.org), and it is available as a pre-built binary package.

- port: `sysutils/py-liquidctl`
- binary: `pkg install py37-liquidctl`
- dependencies: `devel/py-docopt`, `comms/py-hidapi`, `devel/py-pyusb`

liquidctl requires root privileges (doas or sudo) by default.

> Note: as of March 20, 2021, HIDAPI (`comms/hidapi`), [ ... ]

Reviewer: Oh, sorry, `[ ... ]` wasn't a literal suggestion, it just meant "no change from this point forwards".

[bash completions]: extra/completions/liquidctl.bash

> ## Installing on FreeBSD

Reviewer: It seems that DragonFly BSD [has a port too](https://repology.org/project/python:liquidctl/versions), with the same maintainer.

Can you check that your instructions are (at least partially) applicable to it too and, that being true, adjust this section to cover both?

Author: DragonFlyBSD is a separate operating system with its own rules, and I cannot verify its functionality. With that said, I will add a note that it is included in the DragonFly Ports Collection via the FreeBSD Ports Collection and should have matching dependencies. Check for that shortly.

Explanation: Since DragonFlyBSD is a (distant) fork of FreeBSD, its developers are able to run sophisticated scripts to bring the massive FreeBSD Ports Collection to their own OS. That is (almost certainly) why liquidctl is listed as having the same maintainer, even though that person might not even know that his/her work is also being used on DragonFlyBSD (all ports work is done under open BSD licenses).

Reviewer: Oh, i didn't know this process was (possibly) automated, and that the two can vary that substantially.

In that case, "Installing on FreeBSD" is fine.

liquidctl is maintained in the Ports Collection (thanks to ehaupt@FreeBSD.org), and it is available as a pre-built binary package.

> Port:
> 
> `sysutils/py-liquidctl`
> 
> Binary:
> 
> `pkg install py37-liquidctl`
> 
> Dependencies:
> 
> `devel/py-docopt`
> 
> `comms/py-hidapi`
> 
> `devel/py-pyusb`

Reviewer: ```suggestion
- port: `sysutils/py-liquidctl`
- binary: `pkg install py37-liquidctl`
- dependencies: `devel/py-docopt`, `comms/py-hidapi`, `devel/py-pyusb`
```

> liquidctl requires root privileges (doas or sudo) by default.

Reviewer: ```suggestion
By default, root privileges (`doas` or `sudo`) are required to run liquidctl.
```

---

Also, is it possible/advised to change this default behavior?

On Linux this is done with udev rules (see `extra/linux/`).

Author: I _think_ that would best be done with devd(8), of which the GNU userland's udev utility is a (distant) imitation/clone, but I will have to spend a lot of time digging through the man pages to figure out how to get it setup properly for myself before writing a guide -- or else someone who is used to working with such things will need to add that instead.

Until then, I suppose that using `chown` for the file of the USB device is not a downright terrible option, e.g. `sudo chown [user] /dev/ugen[#.#],` as long as it is followed by a careful `chmod 644` (or stricter). But...maybe we should not formally suggest such a bizarre thing. 

I am learning.

Reviewer: It would be interesting to have this on the docs, but it doesn't need to get in right now.  Take your time...

On Linux we're currently using udev to set a `+uaccess` tag, which is pretty much equivalent to `chmod 0666`.  This is fairly common, even though I don't love it; I've recently pinged some package maintainers to discuss how to improve this.

My idea is something like setting the group to `wheel`, or some other common admin group; the issue is that Linux distros tend to have slightly different ways of handling system admin privileges, and some of them are more complicated than just a special group (and in those distros, these groups may not even exist by default).

> NB: As of March 20, 2021, HIDAP (`comms/hidapi`), which is required for `comms/py-hidapi` (and, thus, liquidctl), must be built from the latest port rather than installed as a package, as the package is out of date. Make sure that you have HIDAPI version 0.10.1 or later installed prior to installing `comms/py-hidapi` and liquidctl. Then, liquidctl will work as expected.

Reviewer: ```suggestion
Note: as of March 20, 2021, HIDAPI (`comms/hidapi`), [ ... ]
```

[bash completions]: extra/completions/liquidctl.bash

> ## Installing on FreeBSD or DragonFlyBSD

Reviewer: There's a space in DragonFly BSD.

### FreeBSD

liquidctl is maintained in the FreeBSD Ports Collection (thanks to ehaupt@FreeBSD.org), and it is available as a pre-built binary package.

- port: `sysutils/py-liquidctl`
- binary: `pkg install py37-liquidctl`
- dependencies: `devel/py-docopt`, `comms/py-hidapi`, `devel/py-pyusb`

By default, root privileges (`doas` or `sudo`) are required to run liquidctl.

> To gain full access as a normal user without `doas` or `sudo`, see devd(8) (FreeBSD-specific guide needed). Also, you might consider manually changing the permission of the file of the USB device for an individual session with `chown`, e.g. `sudo chown [user] /dev/ugen[#.#]`.

Reviewer: ```diff
-[...] see devd(8) (FreeBSD-specific guide needed). Also, you [...]
+[...] see `devd(8)`. Alternatively, you [...]
```

Author: Fixed.

> Note: as of March 20, 2021, HIDAP (`comms/hidapi`), which is required for `comms/py-hidapi` (and, thus, liquidctl), must be built from the latest port rather than installed as a package, as the latest package is out of date. Make sure that you have HIDAPI version 0.10.1 or later installed prior to installing `comms/py-hidapi` and liquidctl. Then, liquidctl will work as expected.

Reviewer: ```
HIDAPI
     ^
```

> ### DragonFlyBSD

Reviewer: There's a space in DragonFly BSD.

Author: Good catch. It is not always written that way, probably because everyone (myself included) is used to reading about the 3 oldest established BSDs, none of which use spaces (FreeBSD, OpenBSD, & NetBSD).

> liquidctl is available in DragonFly Ports, having been transferred from the FreeBSD Ports Collection. Installation and use should be the same as in FreeBSD (verification needed).

Reviewer: Do you know _for a fact_ that it was transferred from the FreeBSD collection?  Just because that can happen, doesn't necessarily mean that it happened in this case.

You went to the trouble of thanking ehaupt above... what if ehaupt really did create and test ports for both?

Then, this doesn't really help the user:

```diff
-Installation and use should be the same as in FreeBSD (verification needed).
```

I would simply remove this DragonFly subsubsection (and the heading for the FreeBSD only subsubsection) and just mention DragonFly in the "Installing ..." subsection name.

By the way, you also need to fix the table of contents.

Author: I am 99.9% certain, as it even says in the DragonFly BSD port file itself "$FreeBSD$", etc., etc. But I think that I found a softer way of simply mentioning the DragonFly availability.
